File: frontend/types/order.ts
// Mirrors: app/schemas/order.py, app/schemas/order_item.py, app/schemas/customer_order.py

export interface OrderItem {
  id: number;
  orderId: number;
  productVariantId: number | null;
  quantity: number;
  priceAtPurchase: number;
  createdAt: string;
  updatedAt: string;
}

export interface OrderItemCreate {
  productVariantId: number;
  quantity: number;
  priceAtPurchase: number;
}

export interface OrderItemUpdate {
  quantity?: number;
  priceAtPurchase?: number;
}

export interface Order {
  id: number;
  customerId: number | null;
  status: string | null;
  guestName: string | null;
  guestEmail: string | null;
  guestPhone: string | null;
  shippingAddressId: number | null;
  billingAddressId: number | null;
  couponId: number | null;
  orderStatus: string;
  subtotal: number;
  discountAmount: number;
  shippingAmount: number;
  taxAmount: number;
  totalAmount: number;
  notes: string | null;
  createdAt: string;
  placedAt: string | null;
  items: OrderItem[];
}

export interface OrderCreate {
  customerId?: number | null;
  guestName?: string;
  guestEmail?: string;
  guestPhone?: string;
  orderStatus?: string;
}

export interface OrderUpdate {
  orderStatus?: string;
  notes?: string;
}

export interface OrderStatusUpdate {
  status: string;
}

export interface CustomerOrder {
  id: number;
  customerName: string | null;
  orderStatus: string;
  totalAmount: number;
  createdAt: string;
  items: OrderItem[];
}

export interface ReturnRequestCreate {
  orderItemId: number;
  reason?: string;
}

export interface ReturnRequestStatusUpdate {
  status: string;
  refundAmount?: number;
}

export interface ReturnRequest {
  id: number;
  orderItemId: number;
  customerId: number;
  reason: string | null;
  status: string;
  refundAmount: number | null;
  requestedAt: string;
  processedAt: string | null;
}

File: backend/app/schemas/order.py
from datetime import datetime
from pydantic import BaseModel, ConfigDict, Field, model_validator

from app.schemas.order_item import OrderItemOut


class OrderOut(BaseModel):
    id: int
    customer_id: int | None = None
    status: str | None = None
    guest_name: str | None = None
    guest_email: str | None = None
    guest_phone: str | None = None
    shipping_address_id: int | None = None
    billing_address_id: int | None = None
    coupon_id: int | None = None
    order_status: str
    subtotal: float
    discount_amount: float
    shipping_amount: float
    tax_amount: float
    total_amount: float
    notes: str | None = None
    created_at: datetime
    placed_at: datetime | None = None
    items: list[OrderItemOut] = []

    model_config = ConfigDict(from_attributes=True)

    @model_validator(mode="before")
    @classmethod
    def populate_status_compat(cls, data):
        if isinstance(data, dict):
            if data.get("status") is None and data.get("order_status") is not None:
                data = dict(data)
                data["status"] = data["order_status"]
        return data


class OrderCreate(BaseModel):
    customer_id: int | None = None
    guest_name: str | None = Field(default=None, max_length=100)
    guest_email: str | None = Field(default=None, max_length=255)
    guest_phone: str | None = Field(default=None, max_length=20)
    order_status: str = Field(default="open", max_length=20)


class OrderUpdate(BaseModel):
    order_status: str | None = Field(default=None, max_length=20)
    notes: str | None = None


class OrderStatusUpdate(BaseModel):
    status: str = Field(max_length=20)